jax/interpreters/sharded_jit.py

```diff
@@ -96,24 +112,59 @@ def _sharded_callable(
 
   num_partitions = pxla.reconcile_num_partitions(jaxpr, num_partitions)
   assert num_partitions is not None
-  if num_partitions > xb.local_device_count():
+  if num_partitions > xb.device_count():
     raise ValueError(
         f"sharded_jit computation requires {num_partitions} devices, "
-        f"but only {xb.local_device_count()} devices are available.")
+        f"but only {xb.device_count()} devices are available.")
+  if xb.local_device_count() < num_partitions < xb.device_count():
+    raise NotImplementedError(
+        f"sharded_jit across multiple hosts must use all available devices. "
```

Comment:
The one place in P5X where sharded_jit not inside a pmap might be useful on its own (the init function) would require allowing pod sharded_jit on a subset of global devices (since we don't have partial replication support). But in practice it's even better to just use pmap(sharded_jit) for that too, so that we don't need a separate step that copies the single initialized replica (or single initialized replica per replica set) onto the devices in the other replicas. So I think this restriction is fine.

Reply:
We can also lift this without too much trouble if/when we need it, we just don't have something analogous to `devices` for sharded_jit yet.